tephra-core/src/test/java/co/cask/tephra/TransactionServiceMainTest.java

```diff
@@ -58,6 +60,8 @@ public void run() {
 
       try {
         t.start();
+        // Wait for service to startup
+        TimeUnit.SECONDS.sleep(5);
```

Comment:
Fixed it by using count down latch. Also, TransactionServiceMain's start method was not returning after start-up. Changed it to return after start method is done, to comply with [Apache Commons Daemon spec] (http://commons.apache.org/proper/commons-daemon/apidocs/org/apache/commons/daemon/Daemon.html#start%28%29).